Parsers/ASimNetworkSession/Parsers/vimNetworkSessionMicrosoftSysmon.yaml

```diff
@@ -0,0 +1,336 @@
+        DstHostname = DestinationHostname
+      | where 
+        (array_length(srcipaddr_has_any_prefix)==0 
+                      or has_any_ipv4_prefix(SrcIpAddr,srcipaddr_has_any_prefix)
+        )
+      ;    
+    let Sysmon3_NetworkWindowsEvent =    
+      WindowsEvent
+      | where (isnull(starttime) or TimeGenerated>=starttime) and (isnull(endtime) or TimeGenerated<=endtime) 
+      | where not(disabled)
+      | where Provider == "Microsoft-Windows-Sysmon" and EventID == 3
+    // ***************  Prefilterring *****************************************************************
+      | where  (array_length(dvcaction) ==0  )     /// if filtered by action return nothing
+            and (array_length(ip_any)==0 
+                  or has_any_ipv4_prefix(EventData,ip_any)
+                ) 
+           and (array_length(hostname_has_any)==0 
+                  or EventData has_any(hostname_has_any))  
+            and (isnull(dstportnumber) or EventData has (tostring(dstportnumber))) 
+    // ***************  Prefilterring *****************************************************************   
+      | extend
+            SrcIpAddr = tostring(EventData.SourceIp),
+            DstIpAddr = tostring(EventData.DestinationIp),
+            DstHostname = tostring(EventData.DestinationHostname),
+            SrcHostname = tostring(EventData.SrcHostname),
+            Initiated = tobool(EventData.Initiated)
+      | extend 
+        SourceIsInternal = iff (
+          ipv4_is_private(SrcIpAddr) , "true", "false")
+      | extend 
+        DestinationIsInternal = iff (
+          ipv4_is_private(DstIpAddr) , "true", "false")  // this works only on ipv4 and on RFC standard internal Ips
+    ;
+    let EventParser = (T: (EventData: string)) {
+    T 
+    | parse-kv EventData as 
+          (
+          ProcessGuid:string,
+          ProcessId:string,
+          Image:string,
+          User:string,
+          Protocol:string,
+          Initiated:bool,
+          RuleName:string,
+          SourceIsIpv6:bool,
+          SourcePort:int,
+          DestinationIsIpv6:bool,
+          DestinationPort:int,
+          DestinationHostname:string,
+          UtcTime:datetime
+          ) 
+          with (regex=@'<Data Name="(\w+)">{?([^>]*)}?</Data>')
+    | project-away EventData
+    | project-rename
+         SrcPortNumber = SourcePort,
+         DstPortNumber = DestinationPort,
+         EventStartTime = UtcTime
+    | extend
+         NetworkProtocolVersion = iff((SourceIsIpv6 or DestinationIsIpv6), "IPv6", "IPv4"),
+         ProcessGuid = tostring(split(split(ProcessGuid, "{")[-1], "}")[0]),
+         NetworkDirection = iff(Initiated, "Outbound", "Inbound")
+    | project-away
+         SourceIsIpv6, DestinationIsIpv6
+    };
+    let NetworkEvents = 
+    Sysmon3_NetworkEvent
+    | where Source == "Microsoft-Windows-Sysmon" and EventID==3 // we want to ensure we only have the Event table and not the WindowsEvent table
+    | invoke EventParser ()
```

Comment:
No need to change, but there is no real reason for using a parser as a function in this case. You use it only once.